File: Basics_with_JavaScript/1112 While Loop/12 While loop Excercise/07. Moving.js
function vladi(array){

    let wide = Number(array[0]);
    let length = Number(array[1]);
    let height = Number(array[2]);
    let capacity = wide * length * height;
    let moved = 0;
    let isNoSpace = false;

    let index = 3;
    let command = array[index];
    index++;
    
    while (command !== "Done") {
        moved += Number(command);
        if (capacity <= moved) {
            isNoSpace = true;
            break;
        }
        command = array[index];
        index++;
    }
    // if command Done or capacity <= moved
    let diff = Math.abs(moved - capacity);
    
    if (isNoSpace) {
        console.log(`No more free space! You need ${diff} Cubic meters more.`);
    } else {
        console.log(`${diff} Cubic meters left.`);
    }
    
    }
   
    vladi(["10", 
    "10",
    "2",
    "20",
    "20",
    "20",
    "20",
    "122"])
    ;
    vladi(["10", 
    "1",
    "2",
    "4", 
    "6",
    "Done"])
    ;

File: Basics_with_Python/PythonBasicCourse/5_While_Loop/12_While_Loop___Exercise/07._Moving.py
wide = int(input())
length = int(input())
height = int(input())
available_volume = wide * length * height
moved = input()

while True:
    if moved != "Done":
        moved = int(moved)
        available_volume -= moved
        if available_volume <= 0:
            break
        moved = input()
    elif moved == "Done":
        break

if moved == "Done":
    print(f"{available_volume} Cubic meters left.")
else:
    print(f"No more free space! You need {abs(available_volume)} Cubic meters more.")


# wide = int(input())
# length = int(input())
# height = int(input())
# available_volume = wide * length * height
# moved = input()
#
# while moved != "Done":
#     moved = int(moved)
#     available_volume -= moved
#     if available_volume <= 0:
#         break
#     moved = input()
#
# if moved == "Done":
#     print(f"{available_volume} Cubic meters left.")
# else:
#     print(f"No more free space! You need {abs(available_volume)} Cubic meters more.")


# wide = int(input())
# length = int(input())
# height = int(input())
# moved = 0
# available_volume = wide * length * height
# while available_volume > 0:
#     if moved == "Done":
#         break
#     else:
#         moved = int(moved)
#         available_volume -= moved
#         if available_volume <= 0:
#             break
#         moved = input()
# if moved == "Done":
#     print(f"{available_volume} Cubic meters left.")
# else:
#     print(f"No more free space! You need {abs(available_volume)} Cubic meters more.")
